.github/workflows/deploy.yml

```diff
@@ -0,0 +1,65 @@
+name: auto deploy
+
+on:
+  push:
+    branches:
+      - dep/front
+      - dep/back
+      - dep/docs
+      - dep/app
+
+env:
+  USERNAME: auto-deploy[bot]
+
+jobs:
+  deploy_frontends:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: 🍺 Check out repository code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          repository: modtree/modtree
+          path: other_repo
+          token: ${{secrets.PUSH_TOKEN}}
+
+      - name: Setup git user
+        run: |
+          git config --global user.name $USERNAME
+          git config --global user.email "$USERNAME@modtree.nus"
+        env:
+          USERNAME: ${{ env.USERNAME }}
+
+      - name: 🚀 Send it.
+        run: |
+          cd other_repo
+          echo "── BRANCH ──────────────"
+          echo ${{ github.ref_name }}
+          echo "────────────────────────"
+          REMOTE=https://$USERNAME:$REPO_KEY@github.com/nguyenvukhang/modtree-deploy.git
+
+          function SEND() {
+            git push $REMOTE || git push --force $REMOTE
+          }
+
+          if [[ ${{ github.ref_name }} == "dep/front" ]]; then
+            echo "[ deploying frontend... ]"
+            git checkout -b frontend-deployment; SEND
+
+          elif [[ ${{ github.ref_name }} == "dep/back" ]]; then
+            echo "[ deploying backend... ]"
+            git checkout -b backend-deployment; SEND
+
+          elif [[ ${{ github.ref_name }} == "dep/docs" ]]; then
+            echo "[ deploying docs... ]"
+            git checkout -b docs-deployment; SEND
+
+          elif [[ ${{ github.ref_name }} == "dep/app" ]]; then
+            echo "[ deploying front + back... ]"
+            git checkout -b frontend-deployment; SEND
+            git checkout -b backend-deployment; SEND
+
+          fi
+        env:
+          USERNAME: ${{ env.USERNAME }}
+          REPO_KEY: ${{ secrets.PUSH_TOKEN }}
```

Comment:
tldr:
there are two branches that trigger deployments:
`dep/app` deploys frontend + backend
`dep/docs` deploys... docs.
database will always be running, so no need for a deployment run